src/bin/pg_autoctl/ipaddr.c

```diff
@@ -200,6 +232,32 @@ countSetBits(unsigned int n)
 }
 
 
+/*
+ * countSetBitsv6 returns how many bits are set (to 1) in an IPv6 address, an
+ * array of 16 unsigned char values. When given a netmask, that's the
+ * prefixlen.
+ */
+static unsigned int
+countSetBitsv6(unsigned char *addr)
+{
+	int i = 0;
+	unsigned int count = 0;
+
+	for(i=0; i<16; i++)
+	{
+		unsigned char n = addr[i];
+
+		while (n)
+		{
+			count += n & 1;
```

Comment:
have you considered using `__builtin_popcount` for counting the bits set? Your code works, however this is such a common operation for the cpu there is a specialised implementation for it.

Reply:
Oh you mean like we had in the 80s in another part of the computing world, with the function `logcount` in Common Lisp? See http://clhs.lisp.se/Body/f_logcou.htm.

Well I didn't consider that because I didn't know about `__builtin_popcount`, thanks for mentioning. Looking into it though, it seems that we don't have to sweat much over it if our goal is performance, as explained in https://lemire.me/blog/2016/05/23/the-surprising-cleverness-of-modern-compilers/. Also, we don't run that loop nearly often enough that we would care, right?

Reply:
the only reason to replace it would be readability. The algorithm is not trivial to read, however, popcount is also not widely known in the community. For performance in this case it is neglectable. Lets leave it like this for now.

Reply:
Yeah I we would have to figure out which version of which C compiler actually has popcount built-in, and if they are all using the same name for the feature, etc. Agree with leaving it as it is.